lib/sda/host_msg.go

```diff
@@ -77,13 +86,24 @@ func (t *Token) Id() TokenID {
 	defer tokenMutex.Unlock()
 	if t.cacheId == TokenID(uuid.Nil) {
 		url := network.NamespaceURL + "token/" + t.EntityListID.String() +
-			t.RoundID.String() + t.ProtoID.String() + t.TreeID.String() +
+			t.RoundID.String() + t.ServiceID.String() + t.ProtoID.String() + t.TreeID.String() +
 			t.TreeNodeID.String()
 		t.cacheId = TokenID(uuid.NewV5(uuid.NamespaceURL, url))
 	}
 	return t.cacheId
 }
 
+func (t *Token) Clone() *Token {
+	t2 := Token{}
```

Comment:
Here the IDs are defined as slices which are copied when doing the `:= *`. In the other example you had pointers, which also got copied, but not the structures pointed to... 
If the `Token` would have things like `Tree *Tree`, then the `:= *` would not work.